networking/ciao-cnci-agent/client.go

```diff
@@ -420,6 +429,39 @@ func discoverUUID() (string, error) {
 	return metaData.UUID, nil
 }
 
+//Rebuild network state from database
+func rebuildNetworkState(db *cnciDatabase) error {
+	var lastError error
+	if db == nil {
+		return nil
+	}
+
+	db.SubnetMap.Lock()
+	defer db.SubnetMap.Unlock()
+	db.PublicIPMap.Lock()
+	defer db.PublicIPMap.Unlock()
+
+	for key, subnet := range db.SubnetMap.m {
+		glog.Infof("Key: %v Subnet: %v", key, subnet)
+		err := addRemoteSubnet(subnet)
+		if err != nil {
+			lastError = err
+			glog.Errorf("rebuildNetworkState: %v", err)
```

Comment:
@rbradford Here I try to make forward progress even on error. This means that the CNCI will try to rebuild itself the best it can even if some of the rebuild may fail. Hence I only return the lastError in this function. In the caller of this function, I am considering making forward progress even if the rebuild fails.